implementations/rust/ockam/ockam_command/src/project/enroll.rs

```diff
@@ -32,33 +35,81 @@ pub struct EnrollCommand {
 
 impl EnrollCommand {
     pub fn run(self, options: CommandGlobalOpts) {
-        node_rpc(rpc, (options, self));
+        node_rpc(
+            |ctx, (opts, cmd)| Runner::new(ctx, opts, cmd).run(),
+            (options, self),
+        );
     }
 }
 
-async fn rpc(mut ctx: Context, (opts, cmd): (CommandGlobalOpts, EnrollCommand)) -> Result<()> {
-    async fn go(ctx: &mut Context, opts: &CommandGlobalOpts, cmd: EnrollCommand) -> Result<()> {
-        let tcp = TcpTransport::create(ctx).await?;
-        let (to, meta) = clean_multiaddr(&cmd.to, &opts.config.get_lookup()).unwrap();
-        let projects_sc = crate::project::util::get_projects_secure_channels_from_config_lookup(
-            ctx,
-            opts,
-            &tcp,
-            &meta,
-            &cmd.cloud_opts.route_to_controller,
-            &cmd.node_opts.api_node,
-        )
-        .await?;
-        let to = crate::project::util::clean_projects_multiaddr(to, projects_sc)?;
+struct Runner {
+    ctx: Context,
+    opts: CommandGlobalOpts,
+    cmd: EnrollCommand,
+}
+
+impl Runner {
+    fn new(ctx: Context, opts: CommandGlobalOpts, cmd: EnrollCommand) -> Self {
+        Self { ctx, opts, cmd }
+    }
 
-        let req = Request::post("/members").body(AddMember::new(cmd.member));
-        let mut rpc = RpcBuilder::new(ctx, opts, &cmd.node_opts.api_node)
+    async fn run(self) -> Result<()> {
+        let tcp = TcpTransport::create(&self.ctx).await?;
+        let map = self.opts.config.get_lookup();
+        let to = if let Some(addr) = authority_addr(&self.cmd.to, &map)? {
```

Comment:
That is fine for now, but I found it potentially misleading that multiaddr starting with `/project/name` may be resolved to either project node or authority node depending on the situation